app/ami.py

```diff
@@ -134,8 +133,8 @@ def terminate_resource(self, dryrun: bool) -> bool:
                     snapshot.delete()  # delete() returns None
                 except Exception as e:
                     print(f'Failure when calling snapshot.delete(): {str(e)}')
-                    snapshots_deleted = False  # set to False and continue attempting to delete remaining Snapshots
-        return snapshots_deleted
+                    return e
```

Comment:
The previous logic was attempting to delete the remaining Snapshots after the exception. With the new logic, if any error occurs, the process will not continue attempting.

It's a design decision for this CRAB - whether we want to stop the process when an exception occurs or to continue attempting to delete the remaining resources. I think the process should be uniform for all resources i.e. instances, ami, snapshot, and volume.

Reply:
I changed the logic since I assume if there is an issue deleting one of the snapshots composing the AMI, the other snapshots would have a similar issue as well, but to be sure, I can tweak the logic to be similar to the previous logic so the process will continue iterating through the remaining snapshots if an error is encountered.

Reply:
Just pushed changes that allow the process to continue even if an error is raised, if there are any errors they will all be included in the final slack message. Verified the pytests pass with those changes 👍🏼 .